tests/external_population_test.go

```diff
@@ -0,0 +1,350 @@
+package tests
+
+import (
+	"context"
+	"crypto/md5"
+	"fmt"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	snapshotv1 "github.com/kubernetes-csi/external-snapshotter/client/v6/apis/volumesnapshot/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	controller "kubevirt.io/containerized-data-importer/pkg/controller/common"
+	dvc "kubevirt.io/containerized-data-importer/pkg/controller/datavolume"
+	"kubevirt.io/containerized-data-importer/tests/framework"
+	"kubevirt.io/containerized-data-importer/tests/utils"
+)
+
+const (
+	samplePopulatorName = "sample-populator"
+	populatorGroupName  = "cdi.sample.populator"
+	populatorAPIVersion = "v1alpha1"
+	populatorKind       = "SamplePopulator"
+	populatorResource   = "samplepopulators"
+	localSCName         = "local"
```

Comment:
I was also considering relying on the "local" storage class being there, thing is,
We don't have this storage class downstream so that test will get skipped.

If you want, you could create a function that returns a non-CSI storage
Something like
- If the default storage class is non-csi, good, just return that (downstream we have non-csi NFS lane for example)
- If local is around, pick that
Otherwise error/skip